### PR TITLE
Change basic constraints as mandatory for SPDM 1.3

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2657

### CODE REVIEW

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 1]
```diff
@@ -739,12 +739,14 @@ static bool libspdm_verify_cert_subject_public_key_info(const uint8_t *cert, siz
  *
  * @param[in]  cert                  Pointer to the DER-encoded certificate data.
  * @param[in]  cert_size             The size of certificate data in bytes.
+ * @param[in]  must_have_basic_constraints  This value indicates whether basic_constraints must be present in the Cert
```
Comment (steven-bellock) [inline library/spdm_crypt_lib/libspdm_crypt_cert.c:742]: `must_have_basic_constraints` -> `need_basic_constraints`

File: library/spdm_crypt_lib/libspdm_crypt_cert.c [hunk 2]
```diff
@@ -739,12 +739,14 @@ static bool libspdm_verify_cert_subject_public_key_info(const uint8_t *cert, siz
...
  *
- * @retval  true   verify pass,two case: 1.basic constraints is not present in cert;
+ * @retval  true   verify pass,two case: 1.basic constraints is not present in cert, when must_have_basic_constraints is flase;
```
Comment (steven-bellock) [inline library/spdm_crypt_lib/libspdm_crypt_cert.c:744]: `flase` -> `false`

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: library/spdm_crypt_lib/libspdm_crypt_cert.c (hunk 1 above)
Comment (Wenxing-hou) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c]: Thanks. I have changed the variable name.

File: library/spdm_crypt_lib/libspdm_crypt_cert.c (hunk 2 above)
Comment (Wenxing-hou) [reply library/spdm_crypt_lib/libspdm_crypt_cert.c]: Thanks.  I have fixed the typo.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**